admissions/utils.py

```diff
@@ -0,0 +1,163 @@
+from django.utils import timezone
+from django.db import transaction
+from common.util import send_email
+from django.utils.translation import ugettext_lazy as _
+from django.conf import settings
+from django.apps import apps
+import csv
+
+
+def get_available_interview_locations(datetime_from=None, datetime_to=None):
+    """
+    Accepts a date range and returns all InterviewLocation instances which are available in the given timespan.
+    If this helper does not return any instances it probably is due to a lack of InterviewLocationAvailability
+    instances with valid date ranges
+    """
+
+    if not (datetime_from or datetime_to):
+        raise ValueError("Arguments cannot be None")
+
+    if datetime_from > datetime_to:
+        raise ValueError(
+            "Invalid date range, datetime_from is greater than datetime_to"
+        )
+
+    # Lazy load model due to circular import errors
+    InterviewLocation = apps.get_model(
+        app_label="admissions", model_name="InterviewLocation"
+    )
+    return InterviewLocation.objects.filter(
+        availability__datetime_from__lte=datetime_from,
+        availability__datetime_to__gte=datetime_to,
+    ).distinct()
+
+
+def generate_interviews_from_schedule(schedule):
+    interview_duration = schedule.default_interview_duration
```

Comment:
Should probably add an option the the template "compensate for lost locations" which adds a digital room whenever we are missing a location from out potential max cap?

Reply:
Could be we wont have the need to such an insane number of interviews

Reply:
Has been noted down in the specification sheet